mdata/cache/chunk_cache.go

```diff
@@ -0,0 +1,96 @@
+package cache
+
+import (
+	"flag"
+	"sync"
+
+	"github.com/raintank/metrictank/iter"
+	"github.com/rakyll/globalconf"
+)
+
+var (
+	maxSize uint64
+)
+
+func ConfigSetup() {
+	flags := flag.NewFlagSet("chunk-cache", flag.ExitOnError)
+	// (1024 ^ 3) * 4 = 4294967296
+	flags.Uint64Var(&maxSize, "max-size", 4294967296, "Maximum size of chunk cache in bytes")
+	globalconf.Register("chunk-cache", flags)
+}
+
+type CacheChunk struct {
+	Ts      uint32
+	Next    uint32
+	Prev    uint32
+	itergen iter.IterGen
+}
+
+type MetricCache struct {
```

Comment:
this name is a bit confusing. we need a more specific name, perhaps `ByMetric` (can't come up with anything better for now)